src/wazuh_db/wdb_integrity.c

```diff
@@ -387,3 +388,62 @@ int wdbi_query_clear(wdb_t * wdb, wdb_component_t component, const char * payloa
     cJSON_Delete(data);
     return retval;
 }
+
+// Calculates SHA1 hash from a NULL terminated string array or strings as parameters
+ int wdbi_sha_calculation(const char ** strings_to_hash, os_sha1 hexdigest, unsigned int count, ...)
```

Comment:
I think it's an elegant approach using **...** to solve this.
But probably, having two different methods, one with **...** and the other with **strings_to_hash** it will be easier to use.

Reply:
Thank you. 
That is a good recommendation and it was implemented.